src/stdio.rs

```diff
@@ -40,13 +40,23 @@ impl Stdio {
     }
 
     /// The child inherits from the corresponding parent descriptor.
+    ///
+    /// NOTE that the stdio fd must be in blocking mode, otherwise
+    /// ssh might not flush all output since it considers
+    /// (`EAGAIN`/`EWOULDBLOCK`) as an error
     pub const fn inherit() -> Self {
         Self(StdioImpl::Inherit)
     }
 }
+/// FromRawFd takes ownership of the fd passed in
+/// and closes the fd on drop.
+///
+/// NOTE that the fd must be in blocking mode, otherwise
+/// ssh might not flush all output since it considers
+/// (`EAGAIN`/`EWOULDBLOCK`) as an error
 impl FromRawFd for Stdio {
     unsafe fn from_raw_fd(fd: RawFd) -> Self {
-        Self(StdioImpl::Fd(OwnedFd::from_raw_fd(fd)))
+        Self(StdioImpl::Fd(OwnedFd::from_raw_fd(fd), false))
```

Comment:
Feels like this should really just be a `BorrowedFd` at this point, no?

Reply:
And then we could also add a `From<OwnedFd>` impl for things that are actually owned.

Reply:
> Feels like this should really just be a BorrowedFd at this point, no?

`BorrowedFd` has a lifetime parameter `'fd`, using it would certainly break the apis.

> And then we could also add a From<OwnedFd> impl for things that are actually owned.

Problem is people already uses `FromRawFd` and they probably expects the fd to be closed.

Reply:
That's a fair point. It does mean that the API is kind of sharp in multiple ways.

Maybe mark this whole `impl` as `#[deprecated]` and add a new inherent method that explicitly documents that it will both close the FD _and_ mark it as blocking? That way, existing code will keep working, but new adopters will be able to get the correct/expected behavior. And then in the next major release, we can remove this impl `FromRawFd` in favor of `From<OwnedFd>`.

Reply:
I've marked the `FromRawFd` impl for `Stdio` as `#[deprecated]`, but it's currently "useless" due to https://github.com/rust-lang/rust/issues/39935

> add a new inherent method that explicitly documents that it will both close the FD and mark it as blocking?

And I've added new method `Stdio::from_raw_fd_owned` for that.